### PR TITLE
docs: make it clear the CLI must be downloaded to use templates

### PR DESCRIPTION
**Pull request by sharkymark**
This PR is in response to Issue #5372 about making it clear that the CLI is the same binary as the server.

The PR also is making it clear how to use the CLI to build templates.

### CODE REVIEW

File: docs/templates.md [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+
+The CLI and the server are the same binary. We did this to encourage virality so
+individuals can start their own Coder deployments.
+
+From your local machine, download the CLI for your operating system from the
+[releases](https://github.com/coder/coder/releases) or run:
+
+```sh
+curl -fsSL https://coder.com/install.sh | sh
+
```
Comment (ghuntley) [inline docs/templates.md:22]:
```suggestion
```

File: docs/templates.md [hunk 2]
```diff
@@ -316,12 +375,15 @@ resource "coder_agent" "main" {
...
 
 ## Troubleshooting templates
 
-Occasionally, you may run into scenarios where a workspace is created, but the agent is not connected. This means the agent or [init script](https://github.com/coder/coder/tree/main/provisionersdk/scripts) has failed on the resource.
+Occasionally, you may run into scenarios where a workspace is created, but the
+agent is not connected. This means the agent or [init
+script](https://github.com/coder/coder/tree/main/provisionersdk/scripts) has
+failed on the resource.
 
 ```sh
```
Comment (ghuntley) [inline docs/templates.md:388]:
```suggestion
```console
```

File: docs/templates.md [hunk 3]
```diff
@@ -228,18 +278,20 @@ resource "kubernetes_pod" "podName" {
...
-You can edit a template using the coder CLI. Only
-[template admins and owners](./admin/users.md) can edit a template.
+You can edit a template using the coder CLI. Only [template admins and
+owners](./admin/users.md) can edit a template.
 
-Using the CLI, login to Coder and run the following command to edit a single template:
+Using the CLI, login to Coder and run the following command to edit a single
+template:
 
 ```sh
```
Comment (ghuntley) [inline docs/templates.md:287]:
```suggestion
```console
```

File: docs/templates.md [hunk 4]
```diff
@@ -228,18 +278,20 @@ resource "kubernetes_pod" "podName" {
...
 coder templates edit <template-name> --description "This is my template"
 ```
 
 Review editable template properties by running `coder templates edit -h`.
 
-Alternatively, you can pull down the template as a tape archive (`.tar`) to your current directory:
+Alternatively, you can pull down the template as a tape archive (`.tar`) to your
+current directory:
 
 ```sh
```
Comment (ghuntley) [inline docs/templates.md:296]:
```suggestion
```console
```

File: docs/templates.md [hunk 5]
```diff
@@ -251,36 +303,39 @@ Then, extract it by running:
 tar -xf file.tar
 ```
 
-Make the changes to your template then run this command from the root of the template folder:
+Make the changes to your template then run this command from the root of the
+template folder:
 
 ```sh
```
Comment (ghuntley) [inline docs/templates.md:309]:
```suggestion
```console
```

**Review by ghuntley — APPROVED** [review]
Some minor nits but thanks for putting together a page update!

We use `console` to highlight a shell session command sequence as it looks to the user (with prompts, not just as the contents of a hypothetical script file). Otherwise, we should migrate to `shell`, not `sh`, each time we are in refactoring.

**Review by ericpaulsen — COMMENTED** [review]
looks great! I just pushed a commit for fixing the dead URLs.

**Review by ericpaulsen — APPROVED** [review]

**Review by bpmct — APPROVED** [review]
LGTM :) Thanks Mark

**Review by ghuntley — APPROVED** [review]
Second pass review

**Pull request merged**